src/containers/OfferContainer/OfferContainer.js

```diff
@@ -11,239 +10,235 @@ import {
+        if (orderId) {
+          reuseSavedOrder(orderId);
+        } else {
+          createOrderHandler();
+        }
+      });
     }
-    return offerDetails &&
-      isOrderCreated &&
-      (isFreeOffer ? isFreeOrderReady : true) ? (
-      <Offer
-        offerDetails={offerDetails}
-        orderDetails={orderDetails}
-        couponProps={{
-          ...couponProps,
-          onSubmit: this.onCouponSubmit
-        }}
-        onPaymentComplete={onSuccess}
-        updatePriceBreakdown={this.updatePriceBreakdown}
-        t={t}
-      />
-    ) : (
-      <StyledLoaderContainer>
-        <Header />
-        <StyledLoaderContent>
-          <Loader />
-        </StyledLoaderContent>
-        <Footer />
-      </StyledLoaderContainer>
+    if (offerId === '') {
+      setErrorMsg('Offer not set');
+    }
+  }, []);
+
+  useEffect(() => {
+    if (
+      orderDetails &&
+      orderDetails.totalPrice === 0 &&
+      !orderDetails.discount.applied
+    ) {
+      setIsOfferFree(true);
+    }
+    if (orderDetails.id) {
+      setIsLoading(false);
+    }
+  }, [orderDetails]);
+
+  useEffect(() => {
+    if (isOfferFree) paymentMethodsHandler();
+  }, [isOfferFree]);
+
+  const errorMapping = err => {
+    const errorTypes = {
+      cannotPurchase: ['Offer is blocked for country'],
+      offerNotExist: [
+        "doesn't exist",
+        'does not exist',
+        'Invalid param offerId',
+        'Offer not set'
+      ],
+      alreadyHaveAccess: ['Access already granted'],
+      generalError: ['Request failed with status code 500'],
+      inactive: ['inactive']
+    };
+    const types = Object.keys(errorTypes);
+    return types.find(type =>
+      errorTypes[type].find(item => item.includes(err) || err.includes(item))
     );
+  };
+  if (errorMsg) {
+    return <ErrorPage type={errorMapping(errorMsg)} />;
   }
-}
 
-OfferContainer.propTypes = {
+  return isLoading ? (
+    <StyledLoaderContainer>
+      <Header />
+      <StyledLoaderContent>
+        <Loader />
+      </StyledLoaderContent>
+      <Footer />
+    </StyledLoaderContainer>
+  ) : (
+    <Offer
+      offerDetails={offerDetails}
+      orderDetails={orderDetails}
+      couponProps={{
+        ...couponDetails,
+        onSubmit: onCouponSubmit
+      }}
+      onPaymentComplete={onSuccess}
+      updatePriceBreakdown={updatedOrder => setOrderDetails(updatedOrder)}
+      t={t}
+    />
+  );
+};
+
+OfferWrapper.propTypes = {
   offerId: PropTypes.string,
   onSuccess: PropTypes.func,
   urlProps: PropTypes.shape({
     location: PropTypes.shape({ search: PropTypes.string })
   }),
   t: PropTypes.func
 };
-OfferContainer.defaultProps = {
+OfferWrapper.defaultProps = {
   offerId: '',
   onSuccess: () => {},
   urlProps: {},
   t: k => k
 };
 
-export { OfferContainer as PureOfferContainer };
-
 // export default withTranslation()(labeling()(OfferContainer));
-export default OfferContainer;
+export default OfferWrapper;
```

Comment:
If thats OfferWrapper then we also can change filename

Reply:
I prefer to keep the old name 'OfferContainer', so I rename the OfferWrapper to OfferContainer